evap/evaluation/management/commands/ts.py

```diff
@@ -0,0 +1,70 @@
+import argparse
+import os
+import subprocess  # nosec
+import sys
+import unittest
+
+from django.core.management.base import BaseCommand
+from django.conf import settings
+from django.test.runner import DiscoverRunner
+
+
+class ExportRunner(DiscoverRunner):
+    test_loader = unittest.TestLoader()
+
+    def __init__(self, **kwargs):
+        super().__init__(**kwargs)
+        self.test_loader.testMethodPrefix = 'generate_fixtures'
+
+
+class Command(BaseCommand):
+    def add_arguments(self, parser: argparse.ArgumentParser):
+        subparsers = parser.add_subparsers(dest='command', required=True)
+        compile_parser = subparsers.add_parser('compile')
+        compile_parser.add_argument(
+            '--watch', action='store_true',
+            help='Watch scripts and recompile when they change.',
+        )
+        subparsers.add_parser('test')
+        subparsers.add_parser('generate_fixtures')
+
+    def handle(self, *args, **options):
+        if options['command'] == 'compile':
+            self.compile(**options)
+        elif options['command'] == 'test':
+            self.test(**options)
+        elif options['command'] == 'generate_fixtures':
+            self.generate_fixtures(**options)
+
+    def run_command(self, command):
+        try:
+            subprocess.run(command, check=True)  # nosec
+        except FileNotFoundError:
+            print(f'Could not find {command[0]} command', file=self.stderr)
+        except KeyboardInterrupt:
+            pass
+        except subprocess.CalledProcessError as error:
+            sys.exit(error.returncode)
+
+    def compile(self, watch, **_options):
+        static_directory = settings.STATICFILES_DIRS[0]
+        command = [
+            'npx',
+            'tsc',
+            '--project',
+            os.path.join(static_directory, 'ts', 'tsconfig.compile.json'),
+        ]
+
+        if watch:
+            command += ['--watch']
+
+        self.run_command(command)
+
+    def test(self, **_options):
+        self.run_command(['npx', 'jest'])
+
+    @staticmethod
+    def generate_fixtures(**_options):
+        # Enable debug mode as otherwise a collectstatic beforehand would be necessary
+        test_runner = ExportRunner(keepdb=True, debug_mode=True)
```

Comment:
I think keepdb throws errors in some circumstances. While [the docs](https://docs.djangoproject.com/en/3.2/ref/django-admin/#cmdoption-test-keepdb) say it applies unapplied migrations, I think it doesn't work when rebasing branches with migrations or checking out an older commit or something like that. so I suppose that if you make this the default, you'll need to reset the test db manually from time to time. Not sure how you'd do that with the way the tests are run here. I suggest you test this before merging. Also, if you want to introduce this for the regular test command, we would have to overwrite it, which we're not doing at the moment.

Reply:
Yeah, it cannot unapply a migration if you are in some git operation and the migration file thus is not in the working tree.

I think, it would be fine to have the command use --keepdb by default and require a special call in order to build the test database from scratch. We will have to overwrite the regular test command, but it should only be a handful of lines(?)

Reply:
Then, I’ll just remove `keepdb=True`. I think, it was only for saving some seconds for circumventing that the db must be freshly built. Ideally, we would separate the Django test database and the test database which is used for page rendering, but this can be done later when needed.